DM-14429: Implement RFC-460: move afw.geom content to new geom package

r-owen opened the pull request.

TallJimbo commented on examples/SConscript:
```diff
@@ -0,0 +1,16 @@
+
+# This is more complicated than an examples SConscript should usually be (most can just use
+# BasicSConscript.examples() with no arguments), because we have a source file that's compiled
+# into both a C++ program and a SWIG module.
```
Pre-existing, but `SWIG` -> `pybind11`.

TallJimbo commented on include/lsst/geom.h:
```diff
@@ -0,0 +1,39 @@
+
+/*
+ * An include file to include the header files for lsst::geom
+ */
```
Do we need this file for any kind of backwards compatibility?  If not, I think we should remove it; having these include-everything headers encourages downstream code to include more than it needs.

r-owen replied:
Many packages import afw/geom.h. I would expect the geom package to be small enough that this is a reasonable thing to do, but I can remove it if you feel very strongly about it.

TallJimbo replied:
I agree that it's less of a problem for this package than it is for others.  If other packages do use it, by all means leave it in.

r-owen commented:
I will modernize the copyrights, add a sphinx doc build and fix imports.

Regarding whitespace issues in C++ files: I tried to run clang-format on all C++ code but I see I missed a few files and will fix those. Any remaining whiltespace issues are in my opinion not worth fixing; we should live with the output from clang-format.

r-owen commented:
I cleaned up the issues you raised: add sphinx doc support including a preliminary index.rst, modernize copyright, standardize includes and ran clang-format on a few files I missed the first time. Rebased and ready to merge.
